### DISCUSSION TITLE
Creating a new chart type - Pyramid (we are using it for population data)

### DISCUSSION THREAD

**Discussion by pixerize** [open-sdg/open-sdg#1457] [Ideas]
**Is your feature request related to a problem? Please describe.**
A new type of chart - Pyramid

**Describe the solution that you need**
I am working on it for few days, so far I got here: [link](https://pixerize.github.io/populationData-site/1-1-6/)

**Why do you need this feature?**
My country has a need to show population data with this kind of data structure, and I do believe any other country could benefit from the same

What is left to make it perfect:

1) set graph limit to 24 
Solution tried:

```
graph_limit:
 - unit 1971
   minimum 24
   maximum 24 
```

But it is not working, any ideas?
**UPDATE**: Done with custom editing file [link](https://github.com/pixerize/populationData-site/blob/master/_includes/assets/js/indicatorModel.js), line 300

2) Set one color for male, another for female 
**UPDATE**: Done in  [link](https://github.com/pixerize/populationData-site/blob/master/_includes/components/charts/pyramid.html) lines 15-23

3) Remove negative values in chart and table (negative values for males are necessary in order to be on the opposite side when we set stacked graph)
Solution tried: Line 27 [link](https://github.com/pixerize/populationData-site/blob/master/_includes/components/charts/pyramid.html)
**UPDATE**: Chart values are now positive, but still not in Table tab.

4) reverse the order, lover age ranges should start from the bottom not from the top (we are creating a pyramid shape, so it should be wider from the bottom, and narrower as it is getting to the top)
Solution tried: Line 39 at the [link](https://github.com/pixerize/populationData-site/blob/master/_includes/components/charts/pyramid.html).
**UPDATE**: Done  [link](https://github.com/pixerize/populationData-site/blob/master/_includes/components/charts/pyramid.html) line 24

5) Tooltip values should be from the x-axis, and make tooltip bigger, with more data it will be unusable
Solution tried: From lines 17 - 21 at the [link](https://github.com/pixerize/populationData-site/blob/master/_includes/components/charts/pyramid.html).
**UPDATE**: Tooltip removed for now, didn't find the solution

6) Column Year is irrelevant for this indicator because the Unit measures actually are the years. I have set it to 2021, just to populate data, but what can we do to avoid it, or at least hide it from tooltip and table)
**UPDATE**: Hidden with display: false

7) Remove subcategories on the sidebar, because they will not be editable, and rename Unit Measurement to Census
**UPDATE**: Removed with CSS

You can check the main site repo here: [link](https://github.com/pixerize/populationData-site)

Any help will be greatly appreciated.

Joe

**Comment by LucyGwilliamAdmin** [comment]
@pixerize did you end up being able to do this after?

**Reply by pixerize** [reply]
Hi @LucyGwilliamAdmin, I have finished it 90%. The only thing left is to show the age range for each row, on the y-axis - check the [screenshot ](https://prnt.sc/1z4gve2). 

You can check the pyramid in action here: https://pixerize.github.io/populationData-site/1-7-8/

Any suggestion would be great, thank you.

**Comment by WohnungNDS** [comment]
@pixerize well done! good job! Perhaps it may usefull to add a timeslider like in the [map](https://wohnungnds.github.io/IM_Site/1-2-2/).....

**Reply by pixerize** [reply]
Thank you @WohnungNDS ... Wow, timeslider looks awesome!

If you can share some repos or code for it, I will implement it in the pyramid too when I find time for it.

**Comment by WohnungNDS** [comment]
@pixerize i am not familiar with JS... so I can't provide a file. But perhaps you taka a look [here](https://github.com/open-sdg/open-sdg/blob/1.7.0-dev/_includes/assets/js/plugins/leaflet.yearSlider.js)....
